src/app/services/user-deposit.service.ts

```diff
@@ -311,4 +369,84 @@ export class UserDepositService {
             shareReplay({ bufferSize: 1, refCount: true })
         );
     }
+
+    private getWithdrawPlanObservable(
+        userDepositContract$: Observable<Contract>
+    ): Observable<WithdrawPlan> {
+        return this.withdrawPlanSubject.pipe(
+            switchMapTo(
+                zip(this.raidenService.raidenAddress$, userDepositContract$)
+            ),
+            switchMap(([raidenAddress, userDepositContract]) =>
+                from<Promise<{ amount: string; withdraw_block: string }>>(
+                    userDepositContract.methods
+                        .withdraw_plans(raidenAddress)
+                        .call()
+                )
+            ),
+            map(
+                ({ amount, withdraw_block }) =>
+                    ({
+                        amount: new BigNumber(amount),
+                        withdrawBlock: Number(withdraw_block),
+                    } as WithdrawPlan)
+            ),
+            retryWhen((errors) =>
+                errors.pipe(switchMapTo(this.raidenService.rpcConnected$))
+            ),
+            shareReplay(1)
+        );
+    }
+
+    private getBlocksUntilWithdrawObservable(
+        withdrawPlan$: Observable<WithdrawPlan>
+    ): Observable<number> {
+        const filteredWithdrawPlan$ = withdrawPlan$.pipe(
+            filter((withdrawPlan) => withdrawPlan.amount.isGreaterThan(0))
+        );
+        const blockNumber$ = defer(() =>
+            from<Promise<number>>(this.raidenConfig.web3.eth.getBlockNumber())
```

Comment:
Nitpick: `defer` already `from` its result, no need to `from` inside. If doing so, it´s a nice practice to mark the function as `async` to ensure it returns an observable, because if for some reason it eventually returns something else like an array or `undefined` (for mocks), you may end up with an observable which doesn´t do what you´d expect.